src/iconv.jl

```diff
@@ -240,9 +244,9 @@ end
 ## Functions to encode/decode strings
 
 encoding_string(::Type{ASCIIString}) = "ASCII"
-encoding_string(::Type{UTF8String}) = "UTF-8"
-encoding_string(::Type{UTF16String}) = "UTF-16LE"
-encoding_string(::Type{UTF32String}) = "UTF-32LE"
+encoding_string(::Type{UTF8String})  = "UTF-8"
+encoding_string(::Type{UTF16String}) = (ENDIAN_BOM == 0x04030201) ? "UTF-16LE" : "UTF-16BE"
```

Comment:
Ah, I wanted to ask you: does the endianness of the data stored by Julia for `UTF16String` change depending on the machine endianness? Unfortunately, we probably don't have access to a bug-endian machine running Julia to test that.